applications/MultilevelMonteCarloApplication/python_scripts/cmlmc_utilities.py

```diff
@@ -0,0 +1,781 @@
+        mesh_refinement_coefficient : coefficient of mesh refinement
+        initial_mesh_size : size of first mesh considered
+        minimum_add_level : minimum number of samples to add if at least one is going to be added
+        k0   : Certainty Parameter 0 rates (confidence in the variance models)
+        k1   : Certainty Parameter 1 rates (confidence in the weak convergence model)
+        r1   : Cost increase first iterations C-MLMC
+        r2   : Cost increase final iterations C-MLMC
+        '''
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "k0" : 0.1,
+            "k1" : 0.1,
+            "r1" : 1.25,
+            "r2" : 1.15,
+            "tol0" : 0.25,
+            "tolF" : 0.1,
+            "cphi" : 1.0,
+            "number_samples_screening" : 25,
+            "Lscreening" : 2,
+            "Lmax" : 4,
+            "mesh_refinement_coefficient" : 2,
+            "initial_mesh_size" : 0.5,
+            "minimum_add_level" : 6.0,
+            "splitting_parameter_max" : 0.9,
+            "splitting_parameter_min" : 0.1
+        }
+        """)
+        self.settings = custom_settings
+        '''warning if initial_mesh_size parameter not set by the user'''
+        if not (self.settings.Has("initial_mesh_size")):
+            print("\n ######## WARNING : initial_mesh_size parameter not set ---> using defalut value 0.5 ########\n")
+        '''validate and assign default parameters'''
+        self.settings.ValidateAndAssignDefaults(default_settings)        
+        '''current_number_levels : number of levels of current iteration'''
+        self.current_number_levels = self.settings["Lscreening"].GetInt()
+        '''previous_number_levels : number of levels of previous iteration'''
+        self.previous_number_levels = None
+        '''number_samples : total number of samples at current iteration'''        
+        self.number_samples = [self.settings["number_samples_screening"].GetInt() for i in range (self.settings["Lscreening"].GetInt()+1)]
+        '''difference_number_samples : difference between number of samples of current and previous iterations'''
+        self.difference_number_samples = None
+        '''previous_number_samples : total number of samples of previous iteration'''
+        self.previous_number_samples = None
+        '''rates_error : dictionary containing the values of the parameters
+        calpha : coefficient of the function maximizing bias
+        alpha  : exponent of the function maximizing bias
+        cbeta  : coefficient of the function maximizing statistical error
+        beta   : exponent of the function maximizing statistical error
+        cgamma : coefficient of the function maximizing cost
+        gamma  : exponent of the function maximizing cost
+        '''
+        self.rates_error = {"calpha":None, "alpha":None, "cbeta":None, "beta":None, "cgamma":None, "gamma":None}
+        '''mesh_parameters : reciprocal of minimal mesh size'''        
```

Comment:
for single line comments, you'd best use the "#" symbol

also (for the future, don't need to change here) you can do something like

~~~~py 
    self.mesh_parameters = []   #size_mesh : minimal mesh size